### PR TITLE
[Gov] Addition of Field Trials Section Introduction

### PR DESCRIPTION
**Pull request by ss8171**
Introductory Info for Field Trials

Open Items
- [ ] Add language around assigning cat herder, assigning SPOCs, roles and responsibilities, etc.
- [ ] Note that Each trial participant gets a SPOC
- [ ] Add approach to that allows opt-in/opt-out of presenting test results by Participant
- [ ] Define where feedback (Participant opted-in for publishing) will be published, maintained, repo'd
- [ ] Define where test results (Participant opted-in for publishing) will be published, maintained, repo'd
- [ ] Collaborate w/CIRV/OPNFV on how to Partner for Field Trial approach
- [ ] Add outline of time-frames to diagram for each stage of Field Trials
- [ ] Create Use Cases (UCs) and Questions for each UC (based on original constructs / criteria)
- [ ] Add questions related to correctness and completeness of X technical component (deploy and validate)
- [ ] Discuss how to add agnostic questions related to whatever cookbook, installation approach, or method the Participant used (i.e. cookbooks, install guides, other) in the deploy process, to see if feedback can be solicited where issues and gaps in the process may occur.

Open Minor Edits
- [ ] Change to "VNF Instantiation - Loaded and Started-Only"
- [ ] Change from Deploy, Config, Test", to "Deploy and Validate"
- [ ] Change 'standards' to 'requirements' across document

### CODE REVIEW

**Comment by jbeltranatl** [comment]
Transferred @mtnskiier comments to this PR. @ss8171 https://github.com/cntt-n/CNTT/pull/1214 can be closed as obsolete.

Jim's Comments
"Has OPNFV agreed to receive and coordinate the feedback from field trials? Perhaps it's all contained within the CIRV project in OPNFV?

It would be nice if the diagram had timeframes included. Beta programs have time/value and it will take some coordinated effort to get the multiple users to deploy/test/comment in a "useful" timeframe.

Benchmark data from participants might be too sensitive.

I suspect the only meaningful feedback will be on the cookbook - as I don't imagine anyone is going to inspect the tests or spec that closely.

To get REALLY useful information, we need the participants to deploy AND run some kind of workload on the NFVI - ideally each participant would run something different to stress the NFVI in a different way and uncover unspecified components...

one line in "execute" references Mike Fix - please remove this"

**Comment by collivier** [comment]
@jbeltranatl Of course a few people will inspect the results as we have done in OPNFV for years and as operators are clready doing. Why has it diverged so much (only run the cookbook) from the discussions in RI/RC?

File: doc/gov/chapters/chapter09.md [hunk 1]
```diff
@@ -81,3 +81,321 @@ Exceptions will be clearly recorded in Reference Architectures' Appendices which
...
+<a name="9.5.1"></a>
+
+### 9.5.1 Summary/Field Trials Participants Overview
+
+CNTT has built a Reference Architecture (RA-1) on which Reference Implementation (RI-1) and Reference Conformance (RC-1) requirements have been defined. To ensure value add to Telco industry operators, suppliers, and end user customers, it is running field tests to validate the feasibility, utility, and effectiveness of its requirements and methods (RI-1/RC-1).
+
+<a name="9.5.1.1"></a>
+#### 9.5.1.1 Intents of Field Trials
+
+This is a partnership of CNTT with participants to validate the community is adopting a consistent approach. This is not a VI badging exercise. The trials will validate the RI-1 and the RC-1 test suite requirements and methods themselves, not VNFs or VI systems under test.
```
Comment (collivier) [inline doc/gov/chapters/chapter09.md]: They would rather validate RA1 ad RC1. RI1 should pass RC1 as the OpenStack deployments part of the trials.

File: doc/gov/chapters/chapter09.md [hunk 2]
```diff
@@ -81,3 +81,321 @@ Exceptions will be clearly recorded in Reference Architectures' Appendices which
...
+
+<a name="9.5.1.5"></a>
+#### 9.5.1.5 Expectation 2: Execute the RC-1 Test suite
+
+CNTT will provide the participants with the community RC-1 Test suite, found at this Link: https://github.com/cntt-n/CNTT/blob/master/doc/ref_impl/cntt-ri/chapters/chapter04.md. The participants will execute test cases per instructions and record the quantitative results.
+
+Test case suite should be executed successfully at least 3 times. Three times represents the recommended number of test suite runs to eliminate false positives in results.  A triage process will be used to determine and manage root cause analysis of failures. If the failures are determined to be issues with the participant’s RI, CNTT will convey the issues to the RI and make available SMEs to assist the participant in resolving the issues. When failures are deemed to be caused by an issue or gap in the RA/RI/RC, the community will work to determine the resolution, and modify the RA/RI/RC accordingly.
```
Comment (collivier) [inline doc/gov/chapters/chapter09.md]: Would it mean that RI1 should be verified 3 times in a row too?
R1 as all OPNFV installers is tested once right after deployment and then reinstalled. 3 times may secure better results but most of the operations are covered several times (see rally_full).

File: doc/gov/chapters/chapter09.md [hunk 3]
```diff
@@ -81,3 +81,321 @@ Exceptions will be clearly recorded in Reference Architectures' Appendices which
...
+
+Once the test case suite execution is successful for 3 consecutive iterations, the participant will provide the data of all iterations (both successful and unsuccessful) to CNTT based on participant privacy expectations (See Expectation #4)
+
+<a name="9.5.1.6"></a>
+#### 9.5.1.6 Expectation #3: The Qualitative Survey
+
+At the conclusion of Test Case iterations, the participant will be asked to complete a qualitative survey of their experience. This survey will be used to measure the feasibility, utility, and effectiveness of the RI-1 specifications, installation/configuration methods and RC-1 Test case efficacy. The survey will use an Agile User Story format. The Table below provides an example of the survey questions:
```
Comment (collivier) [inline doc/gov/chapters/chapter09.md]: The feedbacks should go through the underlying projects implementing Cookbook or test suites. Nothing new by CNTT.

File: doc/gov/chapters/chapter09.md [hunk 4]
```diff
@@ -81,3 +81,321 @@ Exceptions will be clearly recorded in Reference Architectures' Appendices which
...
+<a name="9.5.2.3"></a>
+### 9.5.2.3 Field Trial Purpose
+
+In the truest form, adoption of a standard is an indication of it’s success within an industry. Standards developed must be interactively tested in multiple environments or “trialed” to ensure they are practicable, functional, and operative. Without running trials to validate the CNTT approach, standards may not provide intended value across a sufficient spectrum of participating entities to be widely adopted. 
+
+<a name="9.5.2.4"></a>
+#### 9.5.2.4 Intents
+
+1)	This is a partnership approach to validate CNTT community is adopting a consistent approach
+2)	Validate RI-1 specifications and RC-1 test suite, not VNFs or NFVI’s in the trial
```
Comment (collivier) [inline doc/gov/chapters/chapter09.md]: Regarding the current scope, local deployments won't validate RI-1 but RA1

File: doc/gov/chapters/chapter09.md [hunk 5]
```diff
@@ -81,3 +81,321 @@ Exceptions will be clearly recorded in Reference Architectures' Appendices which
...
+
+The objective of Execute Stage is participants implementing field trials tasks and record/assess outcomes CNTT will assemble the Trials team to fully develop the action plan including resource assignments, materials requirements, and timelines. 
+
+Activities include the deployment and configuration of RI-1 and execution of the RC-1 test cases. Vendor community members that commit to the trials will build/setup/prep labs for the trials per the instructions:
+
+1.	Secure appropriate environment space (pre-existing, new build, LaaS)
+
+2.	RI-1 implementation  per published RI-1 Specifications  
+
+3.	RC-1 Test suite will be provided to the participants (MF for detail)
```
Comment (collivier) [inline doc/gov/chapters/chapter09.md]: it's already in CNTT RC.
https://cntt-n.github.io/CNTT/doc/ref_cert/lfn/chapters/chapter03.html#3.4

**Comment by rabi-abdel** [comment]
The vendor implementation area in GitHub with appropriate templates has been proposed (https://github.com/cntt-n/CNTT/pull/1305) as per the discussion. it will be good to agree on the template of the VI (Vendor specific Cookbook) and (Trial report).

**Comment by ss8171** [comment]
@rabi-abdel @mtnskiier @nickchase @jbeltranatl  - Please review and provide feedback or approve.

File: doc/gov/chapters/chapter09.md (hunk 2 above)
Comment (ss8171) [reply doc/gov/chapters/chapter09.md]: Verification three times was designated to help reduce the possibility of false positives.  The community should have a conversation to determine the most appropriate number of iterations.

File: doc/gov/chapters/chapter09.md (hunk 1 above)
Comment (ss8171) [reply doc/gov/chapters/chapter09.md]: If the last sentence is replaced with: "The trials will validate that the RC-1 test suite is sufficient to validate that RI-1 conforms to RA-1, and evaluate the process for testing and results submission. VNFs and VIs are not under test." - will that align better?

Comment (collivier) [reply doc/gov/chapters/chapter09.md]: From my understanding RI1 is already continuously verified by RC1 in OPNFV out of the field trial. By running locally RC, we check conformance of third-party NFVI deployments with RA1. The VNF certification part would leverage on RI1 but I don't think it's part of the field trials.

File: doc/gov/chapters/chapter09.md (hunk 2 above)
Comment (collivier) [reply doc/gov/chapters/chapter09.md]: Yes I see your points. The testing part is good and we may not find false positive. Sometimes a few tempest tests are "falsy" failing but it's mostly due to the SUT selected and the hyperthreading in Functest gates (not in place in RC).

File: doc/gov/chapters/chapter09.md (hunk 5 above)
Comment (ss8171) [reply doc/gov/chapters/chapter09.md]: Cedric, completely agree that it is there and very complete. It is evident that great care was taken to create it. The one area of concern is readability of the average System Admin that would be responsible for RI implementation and RC execution.  Others in the community should be engaged to help bring the test suite to a lowest common denominator level.

Comment (pgoyal01) [reply doc/gov/chapters/chapter09.md]: @collivier is functest complete to test all of RA-1 requirements, for example, RBAC access roles/rights. If not, would the CNTT compliance evolve to include other tests beyond functest.

**Comment by rabiabdel** [comment]
@ss8171 approved from my side. once approved by others, it can be merged by the system.

**Comment by rabi-abdel** [comment]
@collivier cab you re-review?

File: doc/gov/chapters/chapter09.md (hunk 1 above)
Comment (pgoyal01) [reply doc/gov/chapters/chapter09.md]: @ss8171 I prefer your rewrite.

File: doc/gov/chapters/chapter09.md [hunk 6]
```diff
@@ -81,3 +81,322 @@ Exceptions will be clearly recorded in Reference Architectures' Appendices which
...
+
+<a name="9.5.1.1"></a>
+#### 9.5.1.1 Intents of Field Trials
+
+This is a partnership of CNTT with participants to validate the community is adopting a consistent approach. This is not a VI badging exercise. The trials will validate the RI-1 and the RC-1 test suite requirements and methods themselves, not VNFs or VI systems under test. 
+
+<a name="9.5.1.2"></a>
+#### 9.5.1.2 Expectations and Assumptions of Field Trials
+
+CNTT expects to exit the trial with either validation of RI-1 and RC-1 or a set of actions to close gaps. Through the community continuous improvement processes, a badging process that is mutually beneficial to operators and suppliers will be defined/refined. CNTT assumes that performance testing is not included in the field trial.
```
Comment (pgoyal01) [inline doc/gov/chapters/chapter09.md]: Sentence issues, "Through the community continuous improvement processes, a ..."

File: doc/gov/chapters/chapter09.md [hunk 7]
```diff
@@ -81,3 +81,322 @@ Exceptions will be clearly recorded in Reference Architectures' Appendices which
...
+
+<a name="9.5.1.3"></a>
+#### 9.5.1.3 Pre-trials activities
+
+Prior to the beginning of field trials, the CNTT community will define an operational plan, secure resource, and provide all designated contact information required to support the trials participants.  As the results of the trails may produce data and information that could be considered sensitive by participants, CNTT will establish standard data set requirements and secure collection methods to ensure participant privacy protection.
```
Comment (pgoyal01) [inline doc/gov/chapters/chapter09.md]: "secure resource" --> "secure resources"

"... support the trials participants" ----- "... support the trial participants"

File: doc/gov/chapters/chapter09.md [hunk 8]
```diff
@@ -81,3 +81,322 @@ Exceptions will be clearly recorded in Reference Architectures' Appendices which
...
+
+<a name="9.5.1.4"></a>
+#### 9.5.1.4 Expectations of Trials Participants
+
+Trials participants will be expected to commit to establishing an RA-1 compliant NFVi through means that are most appropriate for the participant.
```
Comment (pgoyal01) [inline doc/gov/chapters/chapter09.md]: "Trial participants .."

Comment (pgoyal01) [reply doc/gov/chapters/chapter09.md]: CNTT prefers all upper case NFVI

File: doc/gov/chapters/chapter09.md [hunk 9]
```diff
@@ -81,3 +81,322 @@ Exceptions will be clearly recorded in Reference Architectures' Appendices which
...
+c.	Detail level of data that will be published
+
+2) RI-1 Labs are successfully stood up in all target environments
+* Vendor (3rd party)
+*	Community (OPNFV)
+*	LaaS (e.g. UNH)
+
+3) Engaged NVFI vendors successfully configure their infrastructure per RI-1 and run the RC-1 test suite and are able to provide expert feedback
+
+4) Engaged VNF vendors are able to validate that they can instantiate and run rudimentary validation of VNF functionality on more than one conformant NVFI
```
Comment (pgoyal01) [inline doc/gov/chapters/chapter09.md]: Can engaged VNF vendors create their own infrastructure and configure as per RI-1?

Can OpenStack distributors do the same?

Essentially, that the various category of vendors can create their own labs

File: doc/gov/chapters/chapter09.md [hunk 10]
```diff
@@ -81,3 +81,322 @@ Exceptions will be clearly recorded in Reference Architectures' Appendices which
...
+
+<a name="9.5.2.7.3"></a>
+#### 9.5.2.7.3 Test Case Identification
+
+Specific test cases for the field trials will be documented and provided to the participants based upon the CNTT RI-1  and RC-1 work streams requirements. The technical testing methods, procedures and documentation shall be provided by these work streams. 
+
+<a name="9.5.2.7.4"></a>
+#### 9.5.2.7.4 Vendor Solicitation/Commitment
+
+NFVI and VNF vendor members will be solicited for participation in the trials. The vendors will be required to commit fully to the assessment and evaluation  processes. As previously mentioned, additional discussion is needed to define what results data and at what level is acceptable to be shared.
```
Comment (pgoyal01) [inline doc/gov/chapters/chapter09.md]: Why not all vendors -- NFVI, VNF, OpenStack, LaaS, Testing service providers, etc.

File: doc/gov/chapters/chapter09.md (hunk 7 above)
Comment (ss8171) [reply doc/gov/chapters/chapter09.md]: Updates made

File: doc/gov/chapters/chapter09.md (hunk 6 above)
Comment (ss8171) [reply doc/gov/chapters/chapter09.md]: Updated made

File: doc/gov/chapters/chapter09.md (hunk 9 above)
Comment (ss8171) [reply doc/gov/chapters/chapter09.md]: Resolved

File: doc/gov/chapters/chapter09.md (hunk 10 above)
Comment (ss8171) [reply doc/gov/chapters/chapter09.md]: resolved

**Comment by rabiabdel** [comment]
Looks good to me. @collivier can you please re-review?

**Comment by rabi-abdel** [comment]
@ss8171 is this ready to be merged?

**Comment by rabi-abdel** [comment]
I will keep WIP label on this. once you are happy with it, please remove the label and it can then me merged.

**Pull request merged**